### PR TITLE
chore: Add Hotjar integration

### PR DESCRIPTION
**Pull request by 2dpetkov**
- add a build script to inject Hotjar code snippet in all entry points
- add a `build:inject-analytics` npm script, so that the Vercel custom build command limitation of 256 characters is not hit when all scripts are included

### CODE REVIEW

File: scripts/inject-hotjar.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          h._hjSettings={hjid:${HOTJAR_SITE_ID},hjsv:${HOTJAR_VERSION}};
+          a=o.getElementsByTagName('head')[0];
+          r=o.createElement('script');r.async=1;
+          r.src=t+h._hjSettings.hjid+j+h._hjSettings.hjsv;
+          a.appendChild(r);
+      })(window,document,'https://static.hotjar.com/c/hotjar-','.js?sv=');
+  </script>`;
+
+  const injectedHtml = html.replace(
+    '<div id="root"></div>',
```
Comment (2dpetkov) [inline scripts/inject-hotjar.js:41]: the script should be injected in `<head/>`

File: .env.example [hunk 2]
```diff
@@ -13,6 +13,8 @@ VITE_NUMIA_BASE_URL=
 
 AMPLITUDE_API_KEY=
 AMPLITUDE_SERVER_URL=
+HOTJAR_SITE_ID=
+HOTJAR_VERSION=
```
[the remark this answers is not in the document]
Comment (2dpetkov) [reply .env.example:17]: I like that - didn't know if it was an accepted practice for the project.

[the remark this answers is not in the document]
Comment (2dpetkov) [reply .env.example]: @Christopher-Li done :)

File: scripts/README.md [hunk 3]
```diff
@@ -7,6 +7,14 @@ Analytics service used to collect data to help the dYdX Product team make inform
 2. Add API key in Github > Secrets and Variables > Actions as `AMPLITUDE_API_KEY`
 3. In your deploy scripts add `pnpm run build:inject-amplitude` after your pnpm build / vite build command.
 
+### Hotjar
+User behavior tracking service used help the dYdX Product team make informed product decisions.
```
Comment (aforaleka) [inline scripts/README.md:11]:
```suggestion
User behavior tracking service used to help the dYdX Product team make informed product decisions.
```

File: .github/workflows/deploy-testnet.yml [hunk 4]
```diff
@@ -37,6 +37,7 @@ jobs:
         run: |
           pnpm run build
           pnpm run build:inject-amplitude
+          pnpm run build:inject-hotjar
```
Comment (aforaleka) [inline .github/workflows/deploy-testnet.yml:40]: fyi that this workflow is no longer used!

File: scripts/README.md [hunk 5]
```diff
@@ -7,6 +7,14 @@ Analytics service used to collect data to help the dYdX Product team make inform
...
+
+<b>To use with dydxprotocol/v4-web:</b>
+1. Hotjar account with Site Id and Hotjar version.
+2. Add Site Id and Hotjar version in Github > Secrets and Variables > Actions as `HOTJAR_SITE_ID` and `HOTJAR_VERSION`
```
Comment (aforaleka) [inline scripts/README.md:15]: ah the instruction for amplitude was outdated - we can keep this, but in reality the variables only need to be added on vercel

File: .github/workflows/deploy-testnet.yml (hunk 4 above)
Comment (2dpetkov) [reply .github/workflows/deploy-testnet.yml:40]: shall I then drop this change?

**Review by aforaleka — APPROVED** [review]

File: scripts/README.md (hunk 5 above)
Comment (2dpetkov) [reply scripts/README.md:15]: confirming that DOS's prod deployment will have that in Vercel ✅

**Pull request merged**